### ISSUE TITLE
PORT env variable is not working for docker

### ISSUE THREAD

**Issue by alexandrovas** [ngoduykhanh/PowerDNS-Admin#768]
I want to change default port for PDA in docker using environment variable PORT, but it not working.
My docker-compose.yml file:
```
version: "2"
  
services:
  app:
    image: ngoduykhanh/powerdns-admin:latest
    container_name: powerdns_admin
    network_mode: host
    environment:
      - SQLALCHEMY_DATABASE_URI=mysql://admin:password@localhost/pdnsadmin
      - GUNICORN_TIMEOUT=60
      - GUNICORN_WORKERS=2
      - OFFLINE_MODE=False
      - PORT=8000
```

This is logs:
```
Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade  -> 787bdba9e147, Init DB
INFO  [alembic.runtime.migration] Running upgrade 787bdba9e147 -> 59729e468045, Add view column to setting table
INFO  [alembic.runtime.migration] Running upgrade 59729e468045 -> 1274ed462010, Change setting.value data type
INFO  [alembic.runtime.migration] Running upgrade 1274ed462010 -> 4a666113c7bb, Adding Operator Role
INFO  [alembic.runtime.migration] Running upgrade 4a666113c7bb -> 31a4ed468b18, Remove all setting in the DB
INFO  [alembic.runtime.migration] Running upgrade 31a4ed468b18 -> 654298797277, Upgrade DB Schema
INFO  [alembic.runtime.migration] Running upgrade 654298797277 -> 0fb6d23a4863, Remove user avatar
INFO  [alembic.runtime.migration] Running upgrade 0fb6d23a4863 -> 856bb94b7040, Add comment column in domain template record table
INFO  [alembic.runtime.migration] Running upgrade 856bb94b7040 -> b0fea72a3f20, Update domain serial columns type
INFO  [alembic.runtime.migration] Running upgrade b0fea72a3f20 -> 3f76448bb6de, Add user.confirmed column
[2020-06-29 15:12:55 +0000] [1] [INFO] Starting gunicorn 20.0.4
[2020-06-29 15:12:55 +0000] [1] [INFO] Listening at: http://0.0.0.0:80 (1)
[2020-06-29 15:12:55 +0000] [1] [INFO] Using worker: sync
[2020-06-29 15:12:55 +0000] [10] [INFO] Booting worker with pid: 10
[2020-06-29 15:12:55 +0000] [11] [INFO] Booting worker with pid: 11
```

**Comment by bryanpedini** [comment]
Docker creates another layer of networking between your host (and its ports) and the container itself, which has another IP on a private network that the daemon uses to filter and forward requests.

This means one thing: like a virtual machine, when you open port 80 on "192.168.12.3" for example, which is an internal network on your system and your own system only, the only way of accessing that port 80 is by directly pointing to the IP address (that, remember, with Docker can, and probably does, change every time you start the container), and only from within the host itself. However, the Docker daemon supports forwarding ports from the host to a container, listening with the process "docker_proxy" and redirecting all traffic to the container, which then is responsible to handle it.

All this to explain that you can even have port 80 on 9thousand different containers, because all have a different IP address, none shares anything with each other besides the host kernel and the hardware resources, and if forwarded all with different ports, all of them can be accessed directly and differently, such as host8080->cont80.

I don't know if that's actually the case, but `- PORT` probably refers to the port used to forward requests, which means that you can (and should) not care about which port is used inside the container itself, because you forward a different port on the host to the container (unless you want to run a plain HTTP webserver directly from the container, then it's as simple as 80->80).

**Comment by ptrcnull** [comment]
@bryanpedini, your case includes only Docker containers with default network configuration, in case when someone wants to run a container on the host network, it becomes an obvious issue, as you cannot change the port.

**Comment by bryanpedini** [comment]
> in case when someone wants to run a container on the host network, it becomes an obvious issue, as you cannot change the port.

True, but in my limited experience I never had one valid reason to exclude a reverse proxy in front of my Docker containers handling HTTPS certificates and spreading the traffic.
All tho you are absolutely right, in my humble opinion I think that "pre-made" containers from the community are just to be accepted with the provided configuration, or changed as you need with either a simple `sed` before deploying, or replacing the config file entirely with one of yours before going public; as you can always change everything it's stored inside the container itself.

Besides, if a `PORT` variable exists it should work, tho I don't see it as vital for an application like PowerDNS-Admin, as you can also run it from source, or containerize the app you run from source, or simply change the configs inside the container itself.

I just wanted to point out that Docker creates a second layer of network topology between the host and the container, just in case it was expected that the container would interface with the internet directly.
All tho that is how I got the first question in the first place, and I may be totally wrong, and the OP might be already a pro who just wanted an easier way to edit Nginx's port without diving into the config files inside the container itself. That is just something I didn't consider in my first message (again, @ptrcnull you're absolutely right to point out that my explaination was referring to the default network only).

**Comment by ttinkr** [comment]
I just stumbled upon this problem too.
The core reason why I want to change the port is that I don't want to run my containers as root. And as such the account running the container is not allowed to open ports < 1025. This is why we need the option to change the listening port inside the container.
PS: I am running this in Kubernetes, but I guess the same thing applies to bare docker as well.

**Comment by bryanpedini** [comment]
I have just tested it, and indeed `PORT` is not working, but it is not mentioned in `Docker/Dockerfile` nor in `Docker/entrypoint.sh` for the boot of the application.

An alternative method would be to use the `BIND_ADDRESS` variable instead, I just tried and it's working flawlessly.
